torch/csrc/autograd/function.h

```diff
@@ -369,6 +370,12 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
   /// Returns the name of the dynamic type of the function, for debugging.
   virtual std::string name() const;
 
+  /// The difference between functions `should_compute_output` and
+  /// `task_should_compute_output`: The first one only considers the links in
+  /// the actual graph, while the second (that should only be called during the
+  /// backward/grad pass) also takes into account the current running graph task
+  /// to trim even more edges that are not needed.
```

Comment:
For this description, could it be more precise to say something like: "the engine already trims the edge, we want make sure that the node right before the trimmed edge does not compute the gradient for that edge"